frontend/src/components/CartProductActions/CartProductActions.test.tsx

```diff
@@ -0,0 +1,211 @@
+import { fireEvent, screen } from "@testing-library/react";
+import { vi } from "vitest";
+import { axe } from "vitest-axe";
+
+import * as CartContext from "@/context/CartContext";
+import { nwProduct } from "@/lib/test/fixtures/products";
+import { renderWithRouter } from "@/lib/test/renderWithRouter";
+
+import { CartProductActions } from "./CartProductActions";
+
+const singleMockProduct = {
+  quantity: 782,
+  product: nwProduct,
+} as CartItem;
+
+describe("Given a user is looking at a product in a supermarket cart", () => {
+  describe("when the user is looking the product actions", () => {
+    it("Then it should have no accessibility violations", async () => {
+      const { container } = renderWithRouter(
+        <CartProductActions cartProductItem={singleMockProduct} />,
+      );
+      const results = await axe(container, {
+        rules: {
+          "color-contrast": { enabled: false },
+        },
+      });
+      expect(results).toHaveNoViolations();
+    });
+  });
+
+  describe("When the user views the product actions", () => {
+    it("Then it should display the correct quantity", () => {
+      renderWithRouter(
+        <CartProductActions cartProductItem={singleMockProduct} />,
+      );
+
+      const quantityInput = screen.getByTestId(
+        "cart-product-input",
+      ) as HTMLInputElement;
+      expect(quantityInput.value).toBe("782");
+    });
+  });
+
+  describe("When the user interacts the product action buttons", () => {
+    it("Then using the manual input should update the cart total", async () => {
```

Comment:
💅 Generally best not to use 'should' in tests. It doesn't work if it does not. So it 'does' (pass) or does not (fail). That is what the test is checking. That it does :) 

```suggestion
    it("Then using the manual input updates the cart total", async () => {
```

💭 Also it is not clear what `manual input` means. Typing? Using the arrows on the input? Should be abe to know without having to look at details of the test.